src/reflect/scala/reflect/internal/tpe/GlbLubs.scala

```diff
@@ -167,7 +167,14 @@ private[internal] trait GlbLubs {
       case Nil => Nil
     }
 
-    loop(ts)
+    // The order here matters because type variables and
+    // wildcards can act both as subtypes and supertypes.
+    val (ts2, ts1) = ts.partition(_ exists {
```

Comment:
I don't think this is the right solution. We shouldn't be calling `lub`/`glb` with a list of types that includes type variables or wildcards. These test cases are a great help in debugging this.

Reply:
Any idea how we could avoid that?

Reply:
Recalling my experiments here, the current tests exercise this code path, i.e. type variables being inferred by virtue of participating in the `lub/glb`.

The complete solution would probably come with Dotty but in the mean time it's unsatisfactory explaining to people that they have to swap the arguments to `=:=` to make their code compile :smile: (due to scala/bug#10686). How about approximating just the wildcards?

Reply:
I agree we should improve, but I'd like to take a look at the actual code paths that lead here, and how to remove those non-sensical types from the beginning. The type checker already approximates the expected type using wildcards (as `wildPt`), which shares a similar motiviation.

Reply:
One code path is via `ExistentialType.withTypeVars` in subtyping checks when the other type also has type vars. Try this and see a few of the pos partests fail with "trying to do lub/glb of typevar":

```diff
--- a/src/reflect/scala/reflect/internal/tpe/TypeComparers.scala
+++ b/src/reflect/scala/reflect/internal/tpe/TypeComparers.scala
@@ -529,7 +529,7 @@ trait TypeComparers {
         (rt2.parents forall (isSubType(tp1, _, depth))) &&
           (rt2.decls forall (specializesSym(tp1, _, depth)))
       case et2: ExistentialType =>
-        et2.withTypeVars(isSubType(tp1, _, depth), depth) || fourthTry
+        suspendingTypeVars(typeVarsInType(tp1))(et2.withTypeVars(isSubType(tp1, _, depth), depth)) || fourthTry
       case mt2: MethodType =>
         tp1 match {
           case mt1 @ MethodType(params1, res1) =>
```

I'm not sure what is the correct thing to do here. Since we only care if there is any solution to the existential skolems but not what that solution is, if feels unexpected that this would affect the inference of some unrelated type variables.

Reply:
Agreed that those unrelated type vars should not be constrained -- they should act like skolems, I guess. I'm reluctant to change this until I fully understand how this works, which is not likely to happen anymore for 2.13.0, sadly.

Reply:
My initial comment was wrong -- I see now from your ltr/rtl test case that we get into the situation with wildcards in types. I can't think of a better way to make this more consistent, so, let's go with yours.